dask/dataframe/tests/test_dataframe.py

```diff
@@ -5592,7 +5592,7 @@ def test_attrs_series_in_dataframes():
     # the attrs dict for series in a dataframe. Dask uses df.iloc[:0]
     # when creating the _meta dataframe in make_meta_pandas(x, index=None).
     # Should start xpassing when df.iloc works. Remove the xfail then.
-    assert df.A.attrs == ddf.A.attrs
+    assert {"unit": "KG"} == ddf.A.attrs
```

Comment:
This is interesting:

```
df2 = df.copy()
```

removes the attrs from ``df``. Almost certainly a pandas bug

Reply:
Turns out this is not supposed to work and the fact that this worked at all was an accident (it's related to caching in pandas, we should definitely not rely on this especially since this will change with CoW).